src/WooCommerce.php

```diff
@@ -108,14 +108,14 @@ public static function woocommerce_template_loop_product_thumbnail() {
       // load and render all of the product variations.
       $variation_ids = $product->get_visible_children();
       $placeholders = implode(',', array_fill(0, count($variation_ids), '%d'));
-      $attachment_ids = array_merge($attachment_ids, $wpdb->get_col($wpdb->prepare(
+      $attachment_ids = array_unique(array_merge($attachment_ids, $wpdb->get_col($wpdb->prepare(
         "SELECT pm.meta_value AS attachment_id
         FROM wp_posts p
         INNER JOIN wp_postmeta pm ON pm.post_id = p.ID AND pm.meta_key = '_thumbnail_id'
         WHERE p.ID IN ($placeholders)
         ORDER BY p.menu_order ASC",
         $variation_ids
-      )));
+      ))));
```

Comment:
Just as an idea, I wonder if we could improve the legibility of this
```php
$attachment_ids = array_unique(
  array_merge(
    $attachment_ids,
    $wpdb->get_col(
      $wpdb->prepare(
        "SELECT pm.meta_value AS attachment_id
        FROM wp_posts p
        INNER JOIN wp_postmeta pm ON pm.post_id = p.ID AND pm.meta_key = '_thumbnail_id'
        WHERE p.ID IN ($placeholders)
        ORDER BY p.menu_order ASC",
        $variation_ids
      )
    )
  )
);
```

Reply:
Absolutely, sure!